src/combinatorics/simple_graph/density.lean

```diff
@@ -0,0 +1,251 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.basic
+import order.partition.finpartition
+
+/-!
+# Edge density
+
+This file defines the number and density of edges of a relation/graph.
+
+## Main declarations
+
+Between two finsets of vertices,
+* `relation.interedges`: Finset of edges of a relation.
+* `relation.edge_density`: Edge density of a relation.
+* `simple_graph.interedges`: Finset of edges of a graph.
+* `simple_graph.edge_density`: Edge density of a graph.
+-/
+
+open finset
+
+variables {ι κ α β : Type*}
+
+/-! ### Density of a relation -/
+
+namespace relation
+section asymmetric
+variables (r : α → β → Prop) [Π a, decidable_pred (r a)] {s s₁ s₂ : finset α} {t t₁ t₂ : finset β}
```

Comment:
```suggestion
variables (r : α → β → Prop) [decidable_rel r] {s s₁ s₂ : finset α} {t t₁ t₂ : finset β}
```
Does this work?

Reply:
No because `decidable_rel r` only works for `r : α → α → Prop`... Not quite sure why we have this restriction.

Reply:
Huh. Oh well.